README.md

```diff
@@ -73,26 +104,150 @@ The system uses dedicated queues for managing different job phases:
 
-3. **Infrastructure Dependencies**
-   - MongoDB for job management
-   - AWS services (or Localstack for local development):
-     - SQS for queues
-     - S3 for data storage
-     - SNS for alerts
-     - EventBridge for scheduling
+   # For macOS
+   brew install openssl pkg-config gmp python@3.9
+   ```
 
-## 🚀 Installation & Setup
+2. **Install Rust** (Cross-platform)
+
+   ```bash
+   curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+   source ~/.bashrc  # Or source ~/.zshrc for macOS
+   ```
+
+3. **Clone Repository**
+
+   ```bash
+   git clone https://github.com/madara-alliance/madara-orchestrator.git
+   cd madara-orchestrator
+   git submodule update --init
+   ```
+
+4. **Build SNOS**
+
+   ```bash
+   make snos
+   ```
+
+5. **Build Project**
+
+   ```bash
+   cargo build --release
+   ```
+
+### Local Development Setup
+
+1. **Install Docker** (Cross-platform)
+   Follow the official installation guides:
+
+   - [Ubuntu Installation Guide](https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository)
+   - [macOS Installation Guide](https://docs.docker.com/desktop/install/mac-install/)
+
+2. **Install Foundry** (Cross-platform)
+
+   ```bash
+   curl -L https://foundry.paradigm.xyz | bash
+   foundryup
+   ```
+
+3. **Start Local Services**
+
+   ```bash
+   # Start MongoDB
+   docker run -d -p 27017:27017 mongo
+
+   # Start Localstack
+   docker run -d -p 4566:4566 localstack/localstack@sha256:763947722c6c8d33d5fbf7e8d52b4bddec5be35274a0998fdc6176d733375314
+
+   # Start Anvil in a separate terminal
+   anvil --block-time 1
+   ```
+
+4. **Setup Mock Proving Service**
+
+   ```bash
+   # Create a simple HTTP server that always returns 200
+   # Example using Python
+   python3 -m http.server 8888 &
+
+   # Set the mock prover URL in your .env
+   MADARA_ORCHESTRATOR_ATLANTIC_SERVICE_URL=http://localhost:8888
+   ```
+
+5. **Deploy Mock Verifier Contract**
+
+   You can deploy the mock verifier contract in two ways:
+
+   a. **Using Test Implementation**
+
+   - Reference the implementation in `e2e-tests/tests.rs` which contains `AnvilSetup`
+   - This handles the deployment of both the StarkNet core contract and verifier contract
+
+   ```rust
+   // From e2e-tests/tests.rs
+   let anvil_setup = AnvilSetup::new();
+   let (starknet_core_contract_address, verifier_contract_address) = anvil_setup.deploy_contracts().await;
+   ```
+
+   b. **Manual Deployment**
+
+   - Alternatively, you can deploy the mock verifier directly using Foundry
+   - The contract should implement the same interface as used in the tests
+   - Set the deployed address in your environment:
+
+   ```bash
+   MADARA_ORCHESTRATOR_VERIFIER_ADDRESS=<deployed-contract-address>
+   ```
+
+Note: The mock services are intended for development and testing purposes only.
+In production, you'll need to use actual proving services and verifier contracts.
 
 ### Setup Mode
```

Comment:
do we also want to add setting up pathfinder over here?

Reply:
added